react-app/src/library/services/vendor-implementations/vbet/vbet.test.ts

```diff
@@ -855,21 +855,6 @@ describe('VBetservice', () => {
       expect(statusChangeSpy).toHaveBeenCalledWith({ isConnected: true, isConnecting: false });
     });
 
-    it('webhidRequest, connect with previously connected device but label not matched', async () => {
-      const statusChangeSpy = jest.spyOn(vbetService, 'changeConnectionStatus');
-      const requestSpy = (vbetService.requestWebHidPermissions = jest.fn((callback) => {
-        mockReqDeviceList = mockDeviceList1;
-        callback();
-      }));
-      mockDeviceList = mockDeviceList1;
-      await vbetService.connect('random name');
-
-      const devName = vbetService.deviceInfo;
-      expect(devName.ProductName).toBe(mockTestDevName);
-      expect(requestSpy).toHaveBeenCalled();
-      expect(statusChangeSpy).toHaveBeenCalledWith({ isConnected: true, isConnecting: false });
-    });
-
```

Comment:
@JunzheFan I was working on a release to get your work into production and found that this test was failing.  I believe the main issue resides with `findDevice` which is now imported from the VBeT SDK but no matter what I tried, I was unable to make proper progress.  After some time, I looked at the description of the test and thought... is this test necessary?  If you pass in a device with a different name than what is already present and permitted, it shouldn't connect automatically (unless something may be different in place in the VBeT SDK that I am unaware of).  For example, if "Test Label 123" is passed in but a device with the name "VBeT Label Of Tests 123" is present and has permissions, I wouldn't expect it to automatically connect to the latter but perhaps I'm misunderstanding.  So I removed the test and ran coverage check and we were still at 100%

Reply:
Hey, I pulled down this branch and cd to react-app folder and ran `npm run test` command, but I did not see a failing there with this unit test removed or not . Below is the code of findDevice, so if the pass-in label not match the device name returned from the WebHID API, it would trigger the consent prompt again to let user select a device from the list filtered by 340b venderID  
```
export async function findDevice(name: string) {
  const deviceList = await window.navigator.hid.getDevices();
  for (const device of deviceList) {
    if (name.toLowerCase().includes(device.productName.toLowerCase())) {
      return await createDeviceManager(device);
    }
  }
}
```

Reply:
![image](https://github.com/user-attachments/assets/62a26bbf-73fc-4b1a-8f37-38f4b6e943fc)

i get this error when i run it. i did some investigating and attempts to correct that issue (though right now i don't remember what i did) and the next issue i got was "cannot read properties of null (trying to read toLowerCase)" which would be in the `findDevice` function

Reply:
hey I found out the issue was the `mockDeviceList1` array has a second element which is just a empty object that does not have a productName so the code above `device.productName.toLowerCase()` is derefencing a null value, could you just remove that empty object and add the test back in , and I don't think we need to add null chaining check to the productName as according to the webhid spec this will always be a string even if the undelying device has no name defined 
![image](https://github.com/user-attachments/assets/0e280d1b-77ae-4539-9c78-be98e126dbf7)

Reply:
Perfect! That looked to do it, thanks @JunzheFan ! I'll push up the changes now